### DISCUSSION TITLE
Support reusing oauth2 response when running Grafana behind an AWS ALB with authentication

### DISCUSSION THREAD

**Discussion by pmenglund** [grafana/grafana#40825] [General]
If you configure an AWS ALB with oauth2 authentication:
https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html

It will send the redirect to `<host>/oauth2/ipdresponse` which means the Grafana authentication can't reuse the response and log the user in automatically, even if you configure the same oauth2 authentication in Grafana as in the ALB. You need to click the "login with X" to get logged in, but it would be nice to just make it configurable so the `/oauth2/ipdresponse` can be redirected to e.g. `/login/okta`.

URL rewriting isn't possible in the AWS ALB, so the only place to fix this is here.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
